ctype.h missing from puma_http11.c

**Describe the bug**

`#include <ctype.h>` needs to be added to `ext/puma_http11/puma_http11.c`. Without this include compilation of Puma will fail due to error:

```
include the header <ctype.h> or explicitly provide a declaration for 'isspace' 
1 error generated.
```
This seems to be fixed on 5.0.0.beta1 although it won't help since beta doesn't behave nicely and cannot be used in production.

**To Reproduce**
Just try to compile puma.

**Expected behavior**
Succesfully compiled Puma would be expected behaviour :)

**Desktop (please complete the following information):**
MacOS 15.5.5 and MacOS Big Sur
Puma version 4.3.5

The following is what I get when trying to install Puma 4.3.5 in Ruby 2.7.1 under rbenv with RubyGems 3.1.2

```
% sudo gem install puma           
Building native extensions. This could take a while...
ERROR:  Error installing puma:
	ERROR: Failed to build gem native extension.

    current directory: /Users/ryanmorash/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/puma-4.3.5/ext/puma_http11
/Users/ryanmorash/.rbenv/versions/2.7.1/bin/ruby -I /Users/ryanmorash/.rbenv/versions/2.7.1/lib/ruby/2.7.0 -r ./siteconf20200705-62350-nsgo9p.rb extconf.rb
checking for BIO_read() in -lcrypto... yes
checking for SSL_CTX_new() in -lssl... yes
checking for openssl/bio.h... yes
checking for DTLS_method() in openssl/ssl.h... yes
checking for TLS_server_method() in openssl/ssl.h... yes
checking for SSL_CTX_set_min_proto_version in openssl/ssl.h... yes
creating Makefile

current directory: /Users/ryanmorash/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/puma-4.3.5/ext/puma_http11
make "DESTDIR=" clean

current directory: /Users/ryanmorash/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/puma-4.3.5/ext/puma_http11
make "DESTDIR="
compiling http11_parser.c
ext/puma_http11/http11_parser.c:44:18: warning: unused variable 'puma_parser_en_main' [-Wunused-const-variable]
static const int puma_parser_en_main = 1;
                 ^
1 warning generated.
compiling io_buffer.c
compiling mini_ssl.c
mini_ssl.c:145:7: warning: unused variable 'min' [-Wunused-variable]
  int min, ssl_options;
      ^
mini_ssl.c:299:40: warning: function 'raise_error' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
void raise_error(SSL* ssl, int result) {
                                       ^
2 warnings generated.
compiling puma_http11.c
puma_http11.c:203:22: error: implicitly declaring library function 'isspace' with type 'int (int)' [-Werror,-Wimplicit-function-declaration]
  while (vlen > 0 && isspace(value[vlen - 1])) vlen--;
                     ^
puma_http11.c:203:22: note: include the header <ctype.h> or explicitly provide a declaration for 'isspace'
1 error generated.
make: *** [puma_http11.o] Error 1

make failed, exit code 2
```

--- a/History.md
+++ b/History.md
@@ -1,7 +1,7 @@
 ## Master
 
 * Bugfixes
-  * Explicitly include ctype.h to fix compilation warning and build error on MacOS Big Sur (#2304)
+  * Explicitly include ctype.h to fix compilation warning and build error on macOS with Xcode 12 (#2304)
 
 
 ## 4.3.3 and 3.12.4 / 2020-02-28

--- a/History.md
+++ b/History.md
@@ -1,10 +1,7 @@
 ## Master
 
-* Features
-  * Your feature goes here (#Github Number)
-
 * Bugfixes
-  * Your bugfix goes here (#Github Number)
+  * Explicitly include ctype.h to fix compilation warning and build error on MacOS Big Sur (#2304)
 
 
 ## 4.3.3 and 3.12.4 / 2020-02-28

--- a/ext/puma_http11/puma_http11.c
+++ b/ext/puma_http11/puma_http11.c
@@ -10,6 +10,7 @@
 #include "ext_help.h"
 #include <assert.h>
 #include <string.h>
+#include <ctype.h>
 #include "http11_parser.h"
 
 #ifndef MANAGED_STRINGS

